packages/desktop/src/renderer/components/GlobalClock/GlobalClock.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react'
+import { ClockDisplay } from '@components/core/ClockDisplay/ClockDisplay'
+import { clock } from '@renderer/engine'
+
+const onBpmEdit = (bpm: number) => {
+  clock.bpm = bpm
+}
+
+export const GlobalClock = () => {
```

Comment:
I kind of ran into this with my midi work, Feel like it would be good if a packages ui etc lived in the package so all the code stays contained, easy to upgrade/switch package versions.

I tried making a quick plugin system, but feel into an issue with hooks, maybe you might have a better idea for how to get the tsx living in the subpackage. I made a pr into my working midi branch showing the changes, I'll highlight the code of note https://github.com/cale-bradbury/hedron/pull/9